physics/GWD/drag_suite.F90

```diff
@@ -1414,6 +1414,1223 @@ subroutine drag_suite_run(                                           &
+if ( spp_gwd==1 ) then
+  do i = its,im
+    var_stoch(i)   = var(i)   + var(i)*0.75*spp_wts_gwd(i,1)
+    varss_stoch(i) = varss(i) + varss(i)*0.75*spp_wts_gwd(i,1)
+    varmax_ss_stoch(i) = varmax_ss + varmax_ss*0.75*spp_wts_gwd(i,1)
+    varmax_fd_stoch(i) = varmax_fd + varmax_fd*0.75*spp_wts_gwd(i,1)
+  enddo
+else
+  do i = its,im
+    var_stoch(i)   = var(i)
+    varss_stoch(i) = varss(i)
+    varmax_ss_stoch(i) = varmax_ss
+    varmax_fd_stoch(i) = varmax_fd
+  enddo
+endif
+
+!--- calculate length of grid for flow-blocking drag
+!
+do i=1,im
+   delx   = dx(i)
+   dely   = dx(i)
+   dxy4(i,1)  = delx
+   dxy4(i,2)  = dely
+   dxy4(i,3)  = sqrt(delx*delx + dely*dely)
+   dxy4(i,4)  = dxy4(i,3)
+   dxy4p(i,1) = dxy4(i,2)
+   dxy4p(i,2) = dxy4(i,1)
+   dxy4p(i,3) = dxy4(i,4)
+   dxy4p(i,4) = dxy4(i,3)
+   cleff(i) = psl_gwd_dx_factor*(delx+dely)*0.5
+   cleff_ss(i) =  0.1 * max(dxmax_ss,dxy4(i,3))
+!   cleff_ss(i) =  cleff(i)                        ! consider .....
+enddo
+!
+!-----initialize arrays
+!
+   dtaux = 0.0
+   dtauy = 0.0
+   do i = its,im
+     klowtop(i)    = 0
+     kbl(i)        = 0
+   enddo
+!
+   do i = its,im
+     xn(i)         = 0.0
+     yn(i)         = 0.0
+     ubar (i)      = 0.0
+     vbar (i)      = 0.0
+     roll (i)      = 0.0
+     taub (i)      = 0.0
+     oa(i)         = 0.0
+     ol(i)         = 0.0
+     oc(i)         = 0.0
+     oass(i)       = 0.0
+     olss(i)       = 0.0
+     ulow (i)      = 0.0
+     rstoch(i)     = 0.0
+     ldrag(i)      = .false.
+     icrilv(i)     = .false.
+   enddo
+
+   do k = kts,km
+     do i = its,im
+       usqj(i,k) = 0.0
+       bnv2(i,k) = 0.0
+       vtj(i,k)  = 0.0
+       vtk(i,k)  = 0.0
+       taup(i,k) = 0.0
+       taud_ls(i,k) = 0.0
+       taud_bl(i,k) = 0.0
+       dtaux2d(i,k) = 0.0
+       dtauy2d(i,k) = 0.0
+       dtfac(i,k)   = 1.0
+     enddo
+   enddo
+!
+   if ( ldiag_ugwp ) then
+     do i = its,im
+       dusfc_ls(i) = 0.0
+       dvsfc_ls(i) = 0.0
+       dusfc_bl(i) = 0.0
+       dvsfc_bl(i) = 0.0
+       dusfc_ss(i) = 0.0
+       dvsfc_ss(i) = 0.0
+       dusfc_fd(i) = 0.0
+       dvsfc_fd(i) = 0.0
+     enddo
+     do k = kts,km
+       do i = its,im
+         dtaux2d_ls(i,k)= 0.0
+         dtauy2d_ls(i,k)= 0.0
+         dtaux2d_bl(i,k)= 0.0
+         dtauy2d_bl(i,k)= 0.0
+         dtaux2d_ss(i,k)= 0.0
+         dtauy2d_ss(i,k)= 0.0
+         dtaux2d_fd(i,k)= 0.0
+         dtauy2d_fd(i,k)= 0.0
+       enddo
+     enddo
+   endif
+
+   do i = its,im
+     taup(i,km+1) = 0.0
+     xlinv(i)     = 1.0/xl
+     dusfc(i) = 0.0
+     dvsfc(i) = 0.0
+   enddo
+!
+!  initialize array for flow-blocking drag
+!
+   taufb(1:im,1:km+1) = 0.0
+   hmax(1:im) = 0.0
+   komax(1:im) = 0
+   kbmax(1:im) = 0
+   kblk(1:im) = 0
+!
+   do k = kts,km
+     do i = its,im
+       vtj(i,k)  = t1(i,k)  * (1.+fv*q1(i,k))
+       vtk(i,k)  = vtj(i,k) / prslk(i,k)
+       ro(i,k)   = 1./rd * prsl(i,k) / vtj(i,k) ! density kg/m**3
```

Comment:
Recalculating `1/rd` for every grid point and layer is inefficient. Consider defining a new constant and using it instead.

Reply:
defined rd_inv=1/rd and moved it out of the loop